readthedocs/projects/views/private.py

```diff
@@ -347,7 +348,57 @@ def get_form_kwargs(self, step=None):
 
     def get_template_names(self):
         """Return template names based on step name."""
-        return "projects/import_{}.html".format(self.steps.current)
+        return f"projects/import_{self.steps.current}.html"
+
+    def process_step(self, form):
+        # pylint: disable=too-many-nested-blocks
+        if isinstance(form, ProjectBasicsForm):
+            remote_repository = form.cleaned_data.get("remote_repository")
+            if remote_repository and remote_repository.vcs_provider == GITHUB:
+                remote_repository_relations = (
+                    remote_repository.remote_repository_relation.filter(
+                        user=self.request.user,
+                        account__isnull=False,
+                    )
+                    .select_related("account", "user")
+                    .only("user", "account")
+                )
+                for relation in remote_repository_relations:
+                    service = GitHubService(relation.user, relation.account)
+                    session = service.get_session()
+
+                    for yaml in [
+                        ".readthedocs.yaml",
+                        ".readthedocs.yml",
+                        "readthedocs.yaml",
+                        "readthedocs.yml",
+                    ]:
+                        try:
+                            response = session.head(
+                                f"https://api.github.com/repos/{remote_repository.full_name}/contents/{yaml}",
+                                timeout=1,
+                            )
+                            if response.ok:
+                                log.info(
+                                    "Read the Docs YAML file found for this repository.",
+                                    yaml=yaml,
+                                )
+                                messages.success(
+                                    self.request,
+                                    _(
+                                        "We detected a configuration file in your repository; triggering an initial build"
```

Comment:
I actually haven't tried linking inside a toast message yet, but this could be really nice:

```suggestion
                                        "We detected a configuration file in your repository and started your <a href='{{ build_url }}'>project's first build</a>."
```

Reply:
At least I'd use a little different copy:

```suggestion
                                        "We detected a configuration file in your repository and started your project's first build."
```